Add shorthand variants of `update` and `regenerate` that assume the arguments are unchanged

bzinberg opened the pull request:
Adds shorthand variants of `update` and `regenerate` that assume the arguments are unchanged.  I have so far been in this special case most of the time, and this would save a bit of boilerplate.

bzinberg committed:
Add shorthand variant of `update` that assumes the arguments are unchanged

I have so far been in this special case most of the time.

bzinberg commented:
should probably add a unit test to make sure calling the function in the documented way doesn't crash.

marcoct commented:
That's great. I have also wanted this. It would be great to also do the same for `regenerate`, the form of which would take in a trace and a selection.

bzinberg committed:
Add shorthand variant of `regenerate` that assumes the arguments are unchanged

bzinberg committed:
add unit tests for `update` and `regenerate` shorthands

bzinberg commented:
Added the `regenerate` shorthand and unit tests, and I think it's ready to merge.

bzinberg committed:
now that we're using globals, we can simplify

bzinberg committed:
Thanks @ztangent for explaining how to make the `let` version work

Co-authored-by: ztangent <tanqazx@gmail.com>

bzinberg committed:
oy

bzinberg committed:
On third thought [1], we'll continue to only support defining `@gen (static)` functions in top-level statements

[1] https://probcomp.slack.com/archives/C5T7HNLRY/p1602713555118100?thread_ts=1602701467.114500&cid=C5T7HNLRY
